pex/bin/pex.py

```diff
@@ -503,11 +503,18 @@ def make_relative_to_root(path):
   return os.path.normpath(path.format(pex_root=ENV.PEX_ROOT))
 
 
-def main():
+def main(args=None, log_callback=None):
+  if log_callback is not None:
```

Comment:
mutable object was in ref to sys.argv.  I generally think that the slice should still be in testable code it would make a difference for example if you were to change option frameworks to docopt or argparse.  However, that being said its not super critical and I've moved it.

Also removed the log=None default.

Spoke with you offline and you generally seem okay with SystemExit approach.